### PR TITLE
zio: FXOS8700 Driver

### PR DESCRIPTION
**Pull request by teburd**
A basic, but working, fxos8700 driver based on the current sensors driver, no hardware triggers, no software reconfiguration.

### CODE REVIEW

File: include/zio/zio_dev.h [hunk 1]
```diff
@@ -80,14 +79,21 @@ struct zio_buf;
  * @brief Pre-defined channel types with the ability to extend by a driver
  * implementation.
  *
- * Extended values should always be based on ZIO_LAST_CHAN_TYPE with an
+ * Extended values should always be based on ZIO_CHAN_TYPES count with an
  * additive and defined in a publicly available driver header.
+ *
+ * Example: #define MY_CHAN_TYPE (ZIO_CHAN_TYPES+1)
```
Comment (dbkinder) [inline include/zio/zio_dev.h]:
```suggestion
 * Example: \#define MY_CHAN_TYPE (ZIO_CHAN_TYPES+1)
```

File: include/zio/zio_dev.h [hunk 2]
```diff
@@ -129,18 +135,52 @@ struct zio_dev_chan {
...
 };
 
+/**
+ * @brief Pre-defined attribute types with the ability to extend by a driver
+ * implementation.
+ *
+ * Extended values should always be based on ZIO_ATTR_TYPES count with an
+ * additive and defined in a publicly available driver header.
+ *
+ * Example: #define MY_ATTR_TYPE (ZIO_ATTR_TYPES+1)
```
Comment (dbkinder) [inline include/zio/zio_dev.h]:
```suggestion
 * Example: \#define MY_ATTR_TYPE (ZIO_ATTR_TYPES+1)
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
Something is amiss.  The same changes to zio_dev.h are in two PRs (with the same doxygen problem to fix)

**Comment by WilliamGFish** [comment]
Isn't there already a driver for the fxos8700?

**Comment by bfrog** [comment]
> Isn't there already a driver for the fxos8700?

@WilliamGFish yes there is, however this is one of the first devices I'm using to flesh out a new sensor API

**Comment by WilliamGFish** [comment]
Tom,I see, I'm a little torn between the inefficiency and the simplicity of the current API. Do you have more detail as to your planned changes?Billy..On 3 Jul 2019 18:37, Tom Burdick <notifications@github.com> wrote:
Isn't there already a driver for the fxos8700?

@WilliamGFish yes there is, however this is one of the first devices I'm using to flesh out a new sensor API

**Comment by bfrog** [comment]
@WilliamGFish the inefficiency of the current API is the problem I'm attempting to solve

After some more fiddling around though its clear this isn't the API I'm after just yet, so I'll be reworking this again with an updated API idea

**Pull request closed without merging**